File: src/Belajar_SOLID_02_OOP_Inheritance/Java/Multilevel_Inheritance.java
package Belajar_SOLID_02_OOP_Inheritance.Java;

class Animals{
    public void walk(){
        System.out.println(getClass().getSimpleName() + " walk! ");
    }
}

class Carnivore extends Animals{
    public void eat(){
        System.out.println(getClass().getSimpleName() + " eat! ");
    }
}

class Cats extends Carnivore{}

public class Multilevel_Inheritance {
    public static void main(String[] args) {
        Cats kucing = new Cats();
        kucing.eat();
        kucing.walk();
    }
}

File: src/Belajar_SOLID_02_OOP_Inheritance/Kotlin/Multilevel_Inheritance.kt
package Belajar_SOLID_02_OOP_Inheritance.Kotlin

open class Animals{
    fun walk(){
        println("${javaClass.simpleName} walk!")
    }
}

open class Carnivore:Animals(){
    fun eat(){
        println("${javaClass.simpleName} eat")
    }
}

class Cats:Carnivore()

fun main(){
    val kucing = Cats();
    kucing.eat();
    kucing.walk()
}